src/h3lib/lib/h3Index.c

```diff
@@ -779,6 +779,88 @@ void H3_EXPORT(h3ToGeoBoundary)(H3Index h3, GeoBoundary* gb) {
                           H3_EXPORT(h3IsPentagon)(h3), gb);
 }
 
+/**
+ * Returns the max number of possible icosahedron faces an H3 index
+ * may intersect.
+ *
+ * @return int count of faces
+ */
+int H3_EXPORT(maxFaceCount)(H3Index h3) {
+    // a pentagon always intersects 5 faces, a hexagon never intersects more
+    // than 2 (but may only intersect 1)
+    return H3_EXPORT(h3IsPentagon)(h3) ? 5 : 2;
+}
+
+/**
+ * Find all icosahedron faces intersected by a given  H3 index, represented
+ * as integers from 0-19. The array is sparse; since 0 is a valid value,
+ * invalid array values are represented as -1. It is the responsibility of
+ * the caller to filter out invalid values.
+ *
+ * @param h3 The H3 index
+ * @param out Output array. Must be of size maxFaceCount(h3).
+ */
+void H3_EXPORT(h3GetFaces)(H3Index h3, int* out) {
```

Comment:
Do we want this function to return `int` in order to return an error code on invalid `H3Index`s?

Reply:
As noted before, we're really inconsistent on this - most of our similar functions, e.g. `h3ToGeoBoundary`, `h3ToGeo`, `h3GetBaseCell`, etc, do not validate input. I think it's fine to do the same here, for consistency.

My general feeling is that only the caller knows whether input validation is necessary, and they are empowered to add it with `h3IsValid` if checking input validity is more important than performance for their use case.